File: app/src/main/java/com/example/player/a1610aplayerdemo/MainActivity.java
package com.example.player.a1610aplayerdemo;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.view.Window;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import com.example.player.a1610aplayerdemo.home.fragment.Home_Fragment;
import com.example.player.a1610aplayerdemo.member.fragment.Member_Fragment;
import com.example.player.a1610aplayerdemo.study.fragment.Study_Fragment;
import com.example.player.a1610aplayerdemo.user.fragment.User_Fragment;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity implements ViewPager.OnPageChangeListener {
    private ViewPager vp;
    private RadioGroup rg;
    private RadioButton rb1;
    private RadioButton rb2;
    private RadioButton rb3;
    private RadioButton rb4;
    private Main_Adapter adapter;
    private List<Fragment> list = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        supportRequestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.activity_main);
        initView();
        data();
        adapter();
        listener();
    }

    private void initView() {
        vp = (ViewPager) findViewById(R.id.vp);
        rg = (RadioGroup) findViewById(R.id.rg);
        rb1 = (RadioButton) findViewById(R.id.rb1);
        rb2 = (RadioButton) findViewById(R.id.rb2);
        rb3 = (RadioButton) findViewById(R.id.rb3);
        rb4 = (RadioButton) findViewById(R.id.rb4);
        vp.addOnPageChangeListener(this);
    }

    private void data(){
        list.add(new Home_Fragment());
        list.add(new Member_Fragment());
        list.add(new Study_Fragment());
        list.add(new User_Fragment());
    }

    private void adapter(){
        adapter = new Main_Adapter(getSupportFragmentManager(),list);
        vp.setAdapter(adapter);
    }

    private void listener(){
        rg.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                if (R.id.rb1 == checkedId){
                    vp.setCurrentItem(0);
                }
                if (R.id.rb2 == checkedId){
                    vp.setCurrentItem(1);
                }
                if (R.id.rb3 == checkedId){
                    vp.setCurrentItem(2);
                }
                if (R.id.rb4 == checkedId){
                    vp.setCurrentItem(3);
                }
            }
        });
    }

    @Override
    public void onPageScrolled(int i, float v, int i1) {

    }

    @Override
    public void onPageSelected(int i) {
        switch (i){
            case 0:
                rb1.setChecked(true);
                break;
            case 1:
                rb2.setChecked(true);
                break;
            case 2:
                rb3.setChecked(true);
                break;
            case 3:
                rb4.setChecked(true);
                break;
        }
    }

    @Override
    public void onPageScrollStateChanged(int i) {

    }
}

File: app/src/main/java/com/example/player/a1610aplayerdemo/MainActivity.kt
package com.example.player.a1610aplayerdemo

import android.content.res.Resources
import android.graphics.Color
import android.os.Build
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.DisplayMetrics
import android.util.Log
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.RadioButton
import com.example.player.a1610aplayerdemo.bean.Bean
import com.example.player.a1610aplayerdemo.bean.UserBean
import com.example.player.a1610aplayerdemo.main.MainViewPagerAdapter
import com.example.player.a1610aplayerdemo.net.RetrofitCreator
import com.example.player.a1610aplayerdemo.utils.Height
import com.example.player.a1610aplayerdemo.utils.TokenSp
import io.reactivex.Observable
import io.reactivex.Scheduler
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.functions.Function
import io.reactivex.internal.operators.observable.ObservableOnErrorNext
import io.reactivex.schedulers.Schedulers
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.layout_titlebar.*
import kotlin.math.log
import io.reactivex.Observer as Observer

class MainActivity : AppCompatActivity() {
    var linear_main_choice_flag: Boolean = true
    var linear_main_study_flag: Boolean = false
    var linear_main_vip_flag: Boolean = false
    var linear_main_my_flag: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var defaultDisplay = windowManager.defaultDisplay
        var dis: DisplayMetrics = DisplayMetrics()
        defaultDisplay.getMetrics(dis)

        Height.getHeight().display = defaultDisplay

        getUserData()

        main_vp.adapter = MainViewPagerAdapter(supportFragmentManager)

        fun setRadioButtonBg(flag: Boolean, img: ImageView, rb: RadioButton, linear: LinearLayout) {
            if (flag) {
                //白色
                rb.setTextColor(Color.rgb(89, 172, 255))
                img.setColorFilter(Color.rgb(89, 172, 255))
                //蓝色89172255
            } else {
                rb.setTextColor(Color.rgb(192, 192, 192))
                img.setColorFilter(Color.rgb(192, 192, 192))
            }
        }
        fun diao(){
            setRadioButtonBg(linear_main_choice_flag, iv_main_choice, rb_main_choice, linear_main_choice)
            setRadioButtonBg(linear_main_study_flag, iv_main_study, rb_main_study, linear_main_study)
            setRadioButtonBg(linear_main_vip_flag, iv_main_VIP, rb_main_VIP, linear_main_vip)
            setRadioButtonBg(linear_main_my_flag, iv_main_my, rb_main_my, linear_main_my)

        }
        linear_main_choice.setOnClickListener {
            if (!linear_main_choice_flag) {
                linear_main_choice_flag = true
                linear_main_study_flag = false
                linear_main_vip_flag = false
                linear_main_my_flag = false
            }
            diao()
            main_vp.setCurrentItem(0)
        }
        linear_main_study.setOnClickListener {
            if (!linear_main_study_flag) {
                linear_main_choice_flag = false
                linear_main_study_flag = true
                linear_main_vip_flag = false
                linear_main_my_flag = false

            }
            diao()
            main_vp.setCurrentItem(1)

        }
        linear_main_vip.setOnClickListener {
            if (!linear_main_vip_flag) {
                linear_main_choice_flag = false
                linear_main_study_flag = false
                linear_main_vip_flag = true
                linear_main_my_flag = false
            }
            diao()
            main_vp.setCurrentItem(2)

        }
        linear_main_my.setOnClickListener {
            if (!linear_main_my_flag) {
                linear_main_choice_flag = false
                linear_main_study_flag = false
                linear_main_vip_flag = false
                linear_main_my_flag = true
            }
            diao()
            main_vp.setCurrentItem(3)
        }

    }

    private fun getUserData() {
        var sdk:String = ""+Build.VERSION.SDK_INT
        var map = mapOf<String, String>(
            (Pair("DeviceKey", Constant.DeviceKey)), Pair("Android-channel", "guoyun"),
            Pair("Android-VersionCode", "43"), Pair("Tingyun_Process", "true")
//            ,
//            Pair("device", Build.DEVICE), Pair("sdkVersion", sdk),
//            Pair("brand", Build.BOARD), Pair("product", Build.PRODUCT)
        )


        var map1 = mapOf<String, String>(
            (Pair("DeviceKey", Constant.DeviceKey)),
//            Pair("Android-channel", "guoyun"),
//            Pair("Android-VersionCode", "43"), Pair("Tingyun_Process", "true")
//            ,
            Pair("device", Build.DEVICE), Pair("sdkVersion", sdk),
            Pair("brand", Build.BOARD), Pair("product", Build.PRODUCT)
        )

        RetrofitCreator.getInstance().iServiceApi
            .getUser(map,map1)
            .subscribeOn(Schedulers.io())
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe(object :Observer<Bean<UserBean>>{
                override fun onComplete() {

                }

                override fun onSubscribe(d: Disposable) {
                }

                override fun onError(e: Throwable) {
                    Log.d("123", e.toString())
                }

                override fun onNext(t: Bean<UserBean>) {
                    var data:UserBean? = t.data
                    var accessToken = data?.accessToken?:"123"
                    TokenSp.getInstance().saveToken(accessToken)

                    println("456789--------$accessToken")
                }
            })
    }
}
